test/functional/api/autocmd_spec.lua

```diff
@@ -225,6 +226,51 @@ describe('autocmd api', function()
         local aus = meths.get_autocmds { event = "InsertEnter" }
         eq({ { buflocal = false, command = ':echo "1"', event = "InsertEnter", once = false, pattern = "*" } }, aus)
       end)
+
+      it('should work with buffer numbers', function()
```

Comment:
can you add a test w/ `buffer = 0` just to confirm that that works, otherwise I think it looks good. You can merge I think